test/integration/targets/ansible-galaxy-collection/library/setup_collections.py

```diff
@@ -156,7 +156,11 @@ def publish_collection(module, collection):
 
             # Extract the tarfile to sign the MANIFEST.json
             with tarfile.open(collection_path, mode='r') as collection_tar:
-                collection_tar.extractall(path=os.path.join(collection_dir, '%s-%s-%s' % (namespace, name, version)))
+                if hasattr(tarfile, 'data_filter'):
+                    # Remove this check when Python 3.11 is EOL
```

Comment:
Use a deprecation comment here (implemented by our own `deprecated` plugin for `pylint`):

```suggestion
                    # deprecated: description='extractall fallback without filter' python_version='3.11'
```